exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,2329 @@
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.ResetMMP(context.Background(), currencyBTC)
+	if err != nil && !strings.Contains(err.Error(), "MMP disabled") {
+		t.Error(err)
+	}
+	if _, err = d.WSResetMMP(currencyBTC); err != nil && !strings.Contains(err.Error(), "MMP disabled") {
+		t.Error(err)
+	}
+}
+
+func TestSendRFQ(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.SendRFQ(context.Background(), "BTC-PERPETUAL", 1000, order.Buy)
+	if err != nil && !strings.Contains(err.Error(), "not_enough_funds") {
+		t.Error(err)
+	}
+	if _, err = d.WSSendRFQ("BTC-PERPETUAL", 1000, order.Buy); err != nil && !strings.Contains(err.Error(), "not_enough_funds") {
+		t.Error(err)
+	}
+}
+
+func TestSetMMPConfig(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.SetMMPConfig(context.Background(), currencyBTC, 5, 5, 0, 0)
+	if err != nil && !strings.Contains(err.Error(), "MMP disabled") {
+		t.Error(err)
+	}
+	if _, err = d.WSSetMMPConfig(currencyBTC, 5, 5, 0, 0); err != nil && !strings.Contains(err.Error(), "MMP disabled") {
+		t.Error(err)
+	}
+}
+
+func TestGetSettlementHistoryByCurency(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.GetSettlementHistoryByCurency(context.Background(), currencyBTC, "settlement", "", 10, time.Now().Add(-time.Hour))
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = d.WSRetriveSettlementHistoryByCurency(currencyBTC, "settlement", "", 10, time.Now().Add(-time.Hour))
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetSettlementHistoryByInstrument(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := d.GetSettlementHistoryByInstrument(context.Background(), btcPerpInstrument, "settlement", "", 10, time.Now().Add(-time.Hour))
+	if err != nil {
+		t.Error(err)
+	}
+	if _, err = d.WSRetriveSettlementHistoryByInstrument(btcPerpInstrument, "settlement", "", 10, time.Now().Add(-time.Hour)); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitEdit(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or canManipulateRealOrders isnt set correctly")
+	}
+	_, err := d.SubmitEdit(context.Background(), &OrderBuyAndSellParams{OrderID: "incorrectID", Advanced: "", TriggerPrice: 0.001, Price: 100000, Amount: 123})
+	if err != nil && !strings.Contains(err.Error(), "order_not_found") {
+		t.Error(err)
+	}
+	if _, err = d.WSSubmitEdit(&OrderBuyAndSellParams{
+		OrderID:      "incorrectID",
+		Advanced:     "",
+		TriggerPrice: 0.001,
+		Price:        100000,
+		Amount:       123,
+	}); err != nil && !strings.Contains(err.Error(), "order_not_found") {
+		t.Error(err)
+	}
+}
+
+// Combo Books Endpoints
+
+func TestGetComboIDS(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetComboIDS(context.Background(), currencyBTC, "")
+	if err != nil {
+		t.Error(err)
+	}
+	combos, err := d.WSRetriveComboIDS(currencyBTC, "")
```

Comment:
Above call doesn't seem to need a authentication check. Wondering if that is needed in this function.